vmm/src/vm.rs

```diff
@@ -658,6 +658,9 @@ impl Vm {
         self.devices
             .notify_hotplug(HotPlugNotificationType::CPUDevicesChanged)
             .map_err(Error::DeviceManager)?;
+        let mut config = self.config.as_ref().clone();
+        config.cpus.boot_vcpus = desired_vcpus;
+        self.config = Arc::new(config);
```

Comment:
I was torn between this and switching the config to an `Arc<Mutex<>>`  any preference?

Reply:
Well I guess it depends on the design of how the HTTP API is handling requests. If they are serialized, this is safe, but if we can handle multiple requests at the same time, that means we could end up with a resize_cpus and resize_memory at the same time, which would be unfortunate as it would create a race condition (and I guess Rust compiler would complain).
I don't like the entire config to be put under Mutex either, since this means everytime we try to read some information, we need to lock things.
As an alternative approach, we could use a separate lock that only resize_cpus and resize_memory would use, as a sync mechanism to know that it is safe to read from the config and then update it in `self.config`.
Do you think that would work?

Reply:
@sboeuf Long term it won't just be the vCPUs but also memory details, network and block device details as well as VFIO devices so i'm leaning towards adding a Mutex. The config is only rarely accessed so i'm not to worried about lock contention.

Reply:
Yes this is only about configuration, not really the hot path. So yes please go ahead with `Mutex` here. Actually, you can use `RwLock` for more efficiency as most of the time we are reading the configuration.

Reply:
According to https://www.slideshare.net/mitsunorikomatsu/performance-comparison-of-mutex-rwlock-and-atomic-types-in-rust `Mutex` always outperforms `RwLock` on Linux release builds...

Reply:
He states the performance of `RwLock` degrades when the amount of writers increase, but in this case I expect almost only readers. Anyway, not a big deal, go with the solution you prefer :)
But TBH, we should maybe do some code review of our codebase since we have a mix of RwLock and Mutex.